packages/flutter/lib/src/widgets/tap_and_drag_gestures.dart

```diff
@@ -955,7 +934,9 @@ class TapAndDragGestureRecognizer extends OneSequenceGestureRecognizer with _Tap
 
     _wonArenaForPrimaryPointer = true;
 
-    if (_start != null) {
+    // resolve(GestureDisposition.accepted) will be called when the [PointerMoveEvent] has
+    // moved a sufficient global distance.
+    if (_dragState == _DragState.accepted && _start != null) {
```

Comment:
should assert this in this case?:
`assert(currentUp == null)`

Reply:
I wonder if this is still necessary since the recognizer does not wait before declaring victory when the drag threshold is met. Before, there could have been a drag detected and no victory so it was possible that the `PointerUpEvent` came before accepting the drag. Now, since we declare victory immediately when the drag threshold is met, I don't think a `PointerUpEvent` could come before. What do you think?

Reply:
If the `PointerUpEvent` should not come before, we add this reasonable to prevent #122519 regressing. Assertions can be help in diagnosis when problems occur.

Reply:
`if (_dragState == _DragState.accepted && _start != null) ` should be
 ```dart
if (_start != null) {
  assert(_dragState == _DragState.accepted);
...
}
```

right?